### PR TITLE
use secret_connection from tendermint_p2p crate

### PR DESCRIPTION
**Pull request by melekes**
Refs https://github.com/informalsystems/tendermint-rs/issues/691

### CODE REVIEW

File: src/connection/tcp.rs [hunk 1]
```diff
@@ -39,7 +42,15 @@ pub fn open_secret_connection(
     socket.set_read_timeout(Some(timeout))?;
     socket.set_write_timeout(Some(timeout))?;
 
-    let connection = SecretConnection::new(socket, &identity_key, protocol_version)?;
+    let connection = match SecretConnection::new(socket, &identity_key, protocol_version) {
+        Ok(conn) => conn,
+        Err(error) => match error.downcast_ref::<TmError>() {
```
Comment (melekes) [inline src/connection/tcp.rs:47]: this is ugly... should I attempt to rewrite this part to use `eyre`?

Comment (tony-iqlusion) [reply src/connection/tcp.rs:47]: That's fine for now.

This is a tradeoff with using `eyre` to aggregate error types. If you want to handle them, you either need to use downcasting like this to get the concrete type, or you need to return a `Result` with the underlying error enum.

**Comment by tony-iqlusion** [comment]
@melekes can you rebase this now that `tendermint-p2p` v0.17.0-rc3 is out?

**Comment by melekes** [comment]
> @melekes can you rebase this now that `tendermint-p2p` v0.17.0-rc3 is out?

Done

**Comment by tony-iqlusion** [comment]
Codecov failure is unrelated

**Pull request merged**